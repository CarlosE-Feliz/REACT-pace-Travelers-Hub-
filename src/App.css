.flex {
  display: flex;
}

.nav-bar {
  align-items: center;
  justify-content: space-around;
  margin: 10px;
}

a {
  text-decoration: none;
  color: #3896ff;
}

a:focus {
  border-bottom: 3px solid #3895ff56;
}

.ul-nav {
  list-style: none;
  justify-content: space-around;
}

.nav-item {
  margin-left: 20px;
  text-decoration: none;
}

.logo img {
  margin-right: 20px;
}

.header {
  border-bottom: 2px gray solid;
  margin: 0 40px 0 40px;
}

.img-logo {
  width: 60px;
  margin: 5px;
}

.missions-list {
  width: 100%;
  display: flex;
  justify-content: center;
}

table.missions-table {
  table-layout: fixed;
  width: 80%;
  margin-top: 25px;
}

table.missions-table td,
th {
  border: 1px solid gray;
}

th.mission-description {
  width: 50%;
}

h4.mission-table-heading {
  margin-right: 30px;
}

td h4 {
  font-size: 18px;
}

h4.not-a-member,
.active-member {
  color: white;
  border-radius: 2px;
}

h4.not-a-member {
  background-color: gray;
  border: 1px solid gray;
}

h4.active-member {
  background-color: #18a2b8;
  border: 1px solid #18a2b8;
}

button.join-button .leave-button {
  background-color: transparent;
  border-radius: 3px;
  padding: 2px;
}

button.join-button {
  border: 2px solid #343a40;
  color: #343a40;
}

button.leave-button {
  border: 2px solid #dc3645;
  color: #dc3645;
}

td.mission-status,
.mission-button {
  text-align: center;
}

.joined-missions {
  width: 50%;
}

.joined-mission-item {
  border: 1px solid;
  border-radius: 5px;
  margin: 0.1rem;
}

h4.joined-mission-title {
  margin: 2rem;
}
