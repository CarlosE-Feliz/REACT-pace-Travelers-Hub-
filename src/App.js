import './App.css';
import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import logo from './media/planet.png';
import Rocks from './componentR/Rockets';
import SomethingM from './componentM/Missions';
import MyProfile from './componentP/Profile';
import store from './redux/config';

function App() {
  return (
    <Router>
      <div className="header">
        <nav className="nav-bar flex">
          <div className="logo flex">
            <img src={logo} className="img-logo" alt="Logo" />
            <h3>Space Travalers Hub</h3>
          </div>
          <ul className="flex ul-nav">
            <li className="nav-item">
              <Link to="/rockets">Rockets</Link>
            </li>
            <li className="nav-item">
              <Link to="/mission">Missions</Link>
            </li>
            <li className="nav-item">
              <Link to="/profile">My Profile</Link>
            </li>
          </ul>
        </nav>
      </div>
      <Provider store={store}>
        <Switch>
          <Route exact path="/">
            <Rocks />
          </Route>
          <Route path="/mission">
            <SomethingM />
          </Route>
          <Route path="/rockets">
            <Rocks />
          </Route>
          <Route path="/profile">
            <MyProfile />
          </Route>
        </Switch>
      </Provider>
    </Router>
  );
}

export default App;
