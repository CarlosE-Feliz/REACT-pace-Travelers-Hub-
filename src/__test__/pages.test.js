/* eslint-disable no-console */
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/config';

describe('Test routes', () => {
  test('test rockets', async () => {
    render(<Provider store={store}><App /></Provider>);
    fireEvent.click(screen.getByText('Rockets'));
    const button = document.querySelectorAll('.btn');
    console.log(button);
    expect(await screen.getByText('Rockets')).toMatchSnapshot();
  });

  test('test mission', async () => {
    render(<Provider store={store}><App /></Provider>);
    fireEvent.click(screen.getByText('Missions'));
    expect(await screen.getByText('Status')).toMatchSnapshot();
  });

  test('test profile', async () => {
    render(<Provider store={store}><App /></Provider>);
    fireEvent.click(screen.getByText('My Profile'));
    expect(await screen.getByText('My Rockets')).toMatchSnapshot();
  });
});
