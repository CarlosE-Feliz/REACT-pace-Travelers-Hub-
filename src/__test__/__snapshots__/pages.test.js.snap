// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test routes test mission 1`] = `
<h4
  class="mission-table-heading"
>
  Status
</h4>
`;

exports[`Test routes test profile 1`] = `
<h3>
  My Rockets
</h3>
`;

exports[`Test routes test rockets 1`] = `
<a
  aria-current="page"
  class="active"
  href="/rockets"
  style="font-weight: bold; border-bottom: 3px solid #3895ff56;"
>
  Rockets
</a>
`;
